File: ios/RoutineChart/RoutineChart/Core/Domain/Repositories/FamilyRepository.swift
//
//  FamilyRepository.swift
//  RoutineChart
//
//  Repository protocol for Family data access
//

import Foundation

protocol FamilyRepository {
    func create(_ family: Family) async throws
    func get(id: String) async throws -> Family?
    func update(_ family: Family) async throws
    func getAll() async throws -> [Family]
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/repositories/FamilyRepository.kt
package com.HammersTech.RoutineChart.core.domain.repositories

import com.HammersTech.RoutineChart.core.domain.models.Family
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for Family operations
 */
interface FamilyRepository {
    suspend fun create(family: Family)

    suspend fun update(family: Family)

    suspend fun getById(id: String): Family?

    fun observeById(id: String): Flow<Family?>

    suspend fun getFirst(): Family?

    suspend fun getAll(): List<Family>
}
